libvast/src/system/exporter.cpp

```diff
@@ -154,11 +162,15 @@ behavior exporter(stateful_actor<exporter_state>* self, expression expr,
           self->state.results.push_back(std::move(candidate));
         else
           VAST_DEBUG(self, "ignores false positive:", candidate);
-        mask.append_bits(false, candidate.id() - mask.size());
-        mask.append_bit(true);
+        if (sender == self->state.index || sender == self->state.archive) {
+          mask.append_bits(false, candidate.id() - mask.size());
+          mask.append_bit(true);
+        }
+      }
+      if (sender == self->state.index || sender == self->state.archive) {
+        self->state.stats.processed += candidates.size();
+        self->state.unprocessed -= mask;
```

Comment:
I didn't want to interfere with historical queries in case the EXPORTER handles a universal query. And the (un)processed candidates sounded like they relate to the protocol between EXPORTER and ARCHIVE / INDEX in those cases. If I was mistaken, I'll remove the `if`.